service/history/api/describeworkflow/api.go

```diff
@@ -178,13 +179,37 @@ func Invoke(
 				LastIndependentlyAssignedBuildId: ai.GetLastIndependentlyAssignedBuildId(),
 			}
 		}
-		if ai.CancelRequested {
-			p.State = enumspb.PENDING_ACTIVITY_STATE_CANCEL_REQUESTED
-		} else if ai.StartedEventId != common.EmptyEventID {
-			p.State = enumspb.PENDING_ACTIVITY_STATE_STARTED
+
+		p.State = workflow.ActivityState(ai)
+		p.Attempt = ai.Attempt
+		if p.Attempt < 1 {
+			p.Attempt = 1
```

Comment:
Shouldn't the default here be 0? I see that before we only set this to 1 if there was no retry policy.

Reply:
Let me make it exactly like it was before.